In [3]:
import pandas as pd

print("⚡ 重いデータから日本車を救出中...")
df_raw = pd.read_csv('vehicles.csv',
                     usecols=['price', 'year', 'manufacturer', 'model', 'odometer', 'type'],
                     quoting=3,
                     on_bad_lines='skip',
                     engine='python')

jdm_brands = ['toyota', 'honda', 'nissan', 'subaru', 'mazda', 'mitsubishi', 'suzuki', 'lexus']
df_raw['manufacturer'] = df_raw['manufacturer'].str.lower().str.strip('"')
jdm_df = df_raw[df_raw['manufacturer'].isin(jdm_brands)].copy()

jdm_df.to_csv('jdm_clean_data.csv', index=False)
print("✅ 保存完了！これで次からは一瞬で開けます。")

⚡ 重いデータから日本車を救出中...


FileNotFoundError: [Errno 2] No such file or directory: 'vehicles.csv'

In [ ]:
import pandas as pd

df = pd.read_csv('jdm_clean_data.csv')

print(f"再開準備完了！ 日本車データ {len(df)} 件を読み込みました。")

In [ ]:
import pandas as pd

cols = ['price', 'year', 'manufacturer', 'model', 'odometer', 'type', 'state']
df = pd.read_csv('vehicles.csv', usecols=cols)

df_clean = df[
    (df['price'] > 500) &
    (df['price'] < 100000) &
    (df['year'] >= 1980) &
    (df['manufacturer'].notna())
].copy()

jdm_brands = ['toyota', 'honda', 'nissan', 'subaru', 'mazda', 'mitsubishi', 'suzuki', 'lexus']
df_clean['is_japanese'] = df_clean['manufacturer'].str.lower().isin(jdm_brands)

print(f"全体のうち、分析に使える綺麗なデータは {len(df_clean)} 件でした。")
print(f"そのうち、日本車は {df_clean['is_japanese'].sum()} 件見つかりました！")

japanese_summary = df_clean[df_clean['is_japanese']].groupby('manufacturer')['price'].mean().sort_values(ascending=False)
print("\n--- 日本車メーカー別の平均価格 ---")
print(japanese_summary)

In [ ]:
import pandas as pd

file_path = "vehicles.csv"
cols = ['price', 'year', 'manufacturer', 'model', 'odometer', 'type', 'state']

print("巨大データの荒波を突破中...")
df = pd.read_csv(file_path,
                 usecols=cols,
                 on_bad_lines='skip',
                 low_memory=False)

df_clean = df[
    (df['price'] > 500) &
    (df['price'] < 100000) &
    (df['year'] >= 1980) &
    (df['manufacturer'].notna())
].copy()

print(f"✅ 突破成功！ {len(df_clean)} 件のデータを救出しました。")

jdm_brands = ['toyota', 'honda', 'nissan', 'subaru', 'mazda', 'mitsubishi', 'suzuki', 'lexus']
df_clean['is_japanese'] = df_clean['manufacturer'].str.lower().isin(jdm_brands)
print(f"そのうち日本車は {df_clean['is_japanese'].sum()} 件です。")

In [ ]:
import pandas as pd

cols = ['price', 'year', 'manufacturer', 'model', 'odometer', 'type', 'state']
df = pd.read_csv('vehicles.csv', usecols=cols)

df_clean = df[
    (df['price'] > 500) &
    (df['price'] < 100000) &
    (df['year'] >= 1980) &
    (df['manufacturer'].notna())
].copy()

jdm_brands = ['toyota', 'honda', 'nissan', 'subaru', 'mazda', 'mitsubishi', 'suzuki', 'lexus']
df_clean['is_japanese'] = df_clean['manufacturer'].str.lower().isin(jdm_brands)

print(f"全体のうち、分析に使える綺麗なデータは {len(df_clean)} 件でした。")
print(f"そのうち、日本車は {df_clean['is_japanese'].sum()} 件見つかりました！")

japanese_summary = df_clean[df_clean['is_japanese']].groupby('manufacturer')['price'].mean().sort_values(ascending=False)
print("\n--- 日本車メーカー別の平均価格 ---")
print(japanese_summary)

# 1回目: 404エラー（URL切れ） -> ローカル読み込みに変更
# 2回目: ParserError (EOF inside string) -> quoting=3とengine='python'で突破

In [ ]:
import pandas as pd

file_path = "vehicles.csv"
cols = ['price', 'year', 'manufacturer', 'model', 'odometer', 'type', 'state']

print("⚡ 最終防衛ライン突破作戦を開始...")

try:
    df = pd.read_csv(file_path,
                     usecols=cols,
                     quoting=3,
                     on_bad_lines='skip',
                     engine='python')

    df['price'] = pd.to_numeric(df['price'], errors='coerce')
    df['year'] = pd.to_numeric(df['year'], errors='coerce')

    df_clean = df[
        (df['price'] > 500) &
        (df['price'] < 100000) &
        (df['year'] >= 1980) &
        (df['manufacturer'].notna())
    ].copy()

    print(f"✅ 突破成功！！ 救出したデータ件数: {len(df_clean)} 件")

    jdm_brands = ['toyota', 'honda', 'nissan', 'subaru', 'mazda', 'mitsubishi', 'suzuki', 'lexus']
    df_clean['is_japanese'] = df_clean['manufacturer'].str.lower().str.strip('"').isin(jdm_brands)
    print(f"そのうち日本車は {df_clean['is_japanese'].sum()} 件です。")

except Exception as e:
    print(f"❌ 依然としてエラーが発生中: {e}")

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns

jdm_df = df_clean[df_clean['is_japanese']].copy()

yearly_avg = jdm_df.groupby('year')['price'].mean().reset_index()

plt.figure(figsize=(12, 6))

sns.lineplot(data=yearly_avg, x='year', y='price', marker='o', color='red', linewidth=2)

plt.axvline(x=2001, color='blue', linestyle='--', label='25-Year Rule Border')

plt.title('US Market: Japanese Car Price by Year', fontsize=14)
plt.xlabel('Year of Manufacture', fontsize=12)
plt.ylabel('Average Price (USD)', fontsize=12)
plt.grid(True, alpha=0.3)
plt.legend()
plt.show()

In [ ]:
import pandas as pd

print("⚡ 文字コードを切り替えて、再度救出作戦を開始...")

try:
    df_raw = pd.read_csv('vehicles.csv',
                         usecols=['price', 'year', 'manufacturer', 'model', 'odometer', 'type'],
                         quoting=3,
                         on_bad_lines='skip',
                         engine='python',
                         encoding='latin1')

    jdm_brands = ['toyota', 'honda', 'nissan', 'subaru', 'mazda', 'mitsubishi', 'suzuki', 'lexus']
    df_raw['manufacturer'] = df_raw['manufacturer'].astype(str).str.lower().str.strip('"')
    jdm_df = df_raw[df_raw['manufacturer'].isin(jdm_brands)].copy()

    jdm_df.to_csv('jdm_clean_data.csv', index=False)
    print("✅ 完璧です！ Unicodeの壁を突破し、'jdm_clean_data.csv' を保存しました！")

except Exception as e:
    print(f"❌ まだ壁があります: {e}")

In [ ]:
import pandas as pd

file_path = "vehicles.csv"
try:
    df_head = pd.read_csv(file_path, nrows=5)
    print("✅ 読み込み成功！データの最初の5行はこちらです：")
    display(df_head) # もしエラーが出たら print(df_head) に変えてください
except FileNotFoundError:
    print("❌ ファイルが見つかりません。名前が 'vehicles.csv' になっているか確認してね！")
except Exception as e:
    print(f"❌ エラーが発生しました: {e}")

In [ ]:
import pandas as pd

df = pd.read_csv('jdm_clean_data.csv')

jdm_90s = df[(df['year'] >= 1990) & (df['year'] <= 1999)].copy()

model_ranking = jdm_90s.groupby('model')['price'].agg(['mean', 'count']).reset_index()
top_10 = model_ranking[model_ranking['count'] >= 5].sort_values(by='mean', ascending=False)

print("--- 🇺🇸 1990年代 日本車お宝ランキング ---")
print(top_10.head(10))

**「Supra（スープラ）」と「Skyline GTR R32」のツートップ**

 平均5万ドル（約750万円）超え！まさに25年ルールの象徴です。これらは「ただの中古車」ではなく、アメリカでは「投資対象」になっていることが数字から分かります。

**「Land Cruiser（ランクル）」の圧倒的人気**

land cruiser hdj81（80系）や lx 450（レクサス版ランクル）が上位を占めています。特に80系のディーゼルモデル（HDJ81）は、アメリカで新車販売されなかったため、25年ルールで輸入できるようになった今、猛烈な高値で取引されています。

**「Hiace（ハイエース）」のランクイン**

これも面白いですね！日本の働く車が、アメリカでは「クールなバン（Van Life）」として17,000ドル以上で取引されている現実が見えます。

**「Fuso（ふそう）」の謎？**

fuso fh211（三菱ふそうのトラック）がランクインしているのは、アメリカの物流や特装車市場で日本のトラックの信頼性が評価されているリアルな証拠ですね。

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns
supra_df = jdm_90s[jdm_90s['model'].str.contains('supra', case=False, na=False)].copy()
supra_df['odometer'] = pd.to_numeric(supra_df['odometer'],errors='coerce')
plt.figure(figsize=(10, 6))
sns.scatterplot(data=supra_df, x='odometer', y='price', s=100, color='orange', label='Supra')
sns.regplot(data=supra_df, x='odometer', y='price', scatter=False, color='red', line_kws={"label":"Trend"})
plt.title('Legendary Supra: Price vs Mileage in US', fontsize=15)
plt.xlabel('Odometer (Miles)', fontsize=12)
plt.ylabel('Price (USD)', fontsize=12)
plt.grid(True, alpha=0.3)
plt.legend()
plt.show()

**走行距離を無視した高値（Price Floor**）
通常の車は走行距離が10万マイルを超えると価値が急落しますが、スープラ（特にA80型ターボ）は、走行距離に関わらず最低でも数万ドルという「価格の底」があります。

**「過走行＝ボロ」ではない価値**
20万マイル走っていても高値がついている個体は、アメリカで「一生モノ」としてメンテナンスされ、エンジンが積み替えられたり、フルレストアされたりしていることが多いです。データ上の「点」が高い位置にあるのは、その個体が持つストーリーや情熱が価格に乗っているからです。

**投資対象としての安定性**
赤いトレンドライン（傾向線）が意外となだらかではありませんか？これは「距離が伸びても価値が減りにくい（減価償却が遅い）」ことを示しており、アメリカ人がスープラを単なる足車ではなく、**「資産」**として見ている証拠です

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv('jdm_clean_data.csv')

kei_keywords = 'mini truck|kei|hijet|carry|acty|sambar|mini cab'
kei_df = df[df['model'].str.contains(kei_keywords, case=False, na=False)].copy()

kei_trends = kei_df.groupby('year')['price'].mean().reset_index()

plt.figure(figsize=(12,6))
sns.lineplot(data=kei_trends, x='year', y='price', marker='o', color='green')

plt.title('The Rise of Kei Cars in US Market', fontsize=15)
plt.xlabel('Model Year', fontsize=12)
plt.ylabel('Average Price (USD)', fontsize=12)
plt.grid(True, alpha=0.3)
plt.show()

print(f"抽出された軽自動車データ; {len(kei_df)} 件")
print(kei_df[['year', 'model', 'price']].head(10))

**「25年ルール」の壁（1991年〜1992年の高値）**
1991年式のデータが10,000ドル近くなっているのは、アメリカで「25年」を無事に経過し、「クラシックカー」として法的に認められた個体だからです。アメリカのファンはこの「合法性」に高いお金を払います。

**1993年の「谷」の正体**
1993年式のデータが2,000ドル付近まで落ちているのは、おそらく**「まだ日本にいて、これからアメリカに運ばれる予定（輸送費抜き）」**の価格が含まれているか、程度に難がある個体が混ざっているためです。

**事実**: 実際にアメリカのオークションサイト（Bring a Trailerなど）では、状態が良い1993年式のアクティは**8,600ドル（約130万円）**で落札されることもあります。

**「Mini Truck」ブームの証明** 抽出結果を見ると acty minitruck という名前が出てきています。アメリカでは「農作業用の便利な道具（UTV）」としてだけでなく、**「安くてカッコいい日本車」としての需要が急増しており、2024年の軽トラ輸入台数は2018年の4倍以上**に膨れ上がっています。

**3つのレシピの「無限ループ」イメージ**

新しい分析をしたい時は、このサイクルを回すだけです。

**「これ知りたい！」（問い）**

例：1990年代のハイゼット（Hijet）だけはどうなってる？

**レシピ1：絞り込み（Filtering）**

ハイゼットかつ1990年〜1999年に絞る。

**レシピ3：新しい情報（Feature Engineering）**

走行距離を「10万キロ以上か以下か」のラベルを付ける。

**レシピ2：グループ化（Grouping**）

そのラベルごとに平均価格を出す。

**「わかった！」（可視化）**

グラフにする。

In [4]:
kei_4 = df[df['model'].str.contains('acty|hijet|carry|sambar', case=False, na=False)].copy()
plt.figure(figsize=(10, 6))
sns.set_style("whitegrid")

ax = sns.barplot(data=kei_4, x='model', y='price', palette='magma', errorbar=None)

plt.title('Market Value Battle: Japanese Mini Trucks in US', fontsize=16, fontweight='bold')
plt.ylabel('Average Price (USD)', fontsize=12)
plt.xlabel('Model Name', fontsize=12)

plt.show()

print(kei_4.groupby('model')['price'].mean().sort_values(ascending=False))

NameError: name 'df' is not defined

**純粋な「acty」ブランドの強さ**

平均価格 **$8,629** というのは、日本での新車価格に近い（あるいは超える）驚きの数字です。アメリカ人が「軽トラ」という道具としてだけでなく、ホンダの歴史的なJDM車として価値を見出している証拠と言えます。

**名称による価格差の謎**

「minitruck」と付いている方が安いのは、実用車（農場用など）として使い込まれた個体が多いからかもしれません。逆に「acty」とのみ記されているものは、カスタム済みや極上車として扱われている可能性があります。

**1991年式の圧倒的高値**

1枚目の折れ線グラフで、1991年式が $10,000 に到達しているのは、まさに「25年ルール」解禁直後の、最も法的に安定した「クラシックJDM」としてのプレミアム価格ですね。

In [ ]:
df['odometer'] = pd.to_numeric(df['odometer'], errors='coerce')

df['mileage_group'] = 'Under 100k miles'
df.loc[df['odometer'] >= 100000, 'mileage_group'] = 'Over 100k miles'

mileage_analysis = df.groupby('mileage_group')['price'].mean()
print(mileage_analysis)

plt.figure(figsize=(8, 5))
sns.barplot(x=mileage_analysis.index, y=mileage_analysis.values, palette='viridis')
plt.title('Is Japanese Maintenance Powerful? (Price vs Mileage)', fontsize=14)
plt.ylabel('Average Price (USD)')
plt.show()

**10万マイル（16万km）超えでも8,000ドル以上の価値！**

普通の乗用車なら、16万kmを超えるとアメリカでも「廃車寸前」として価格がガクンと落ちます。しかし、軽トラは**平均8,000ドル（約120万円）以上**を維持しています。これはバディの言っていた「日本のメンテナンスへの信頼」の動かぬ証拠です！

**低走行車（Under 100k）は超プレミアム！**

1.9万ドル（約280万円）という数字が出ていますね。新車価格を上回るような勢いです。「日本で大切に乗られていた低走行の軽トラ」は、今やアメリカでは高級車並みの扱いを受けていることが分かります。

In [ ]:
plt.savefig('mileage_analysis.png', bbox_inches='tight', dpi=300)
print("走行距離のグラフを保存しました！")

In [ ]:
print(df.columns)

In [ ]:
df['is_camper'] = df['model'].str.contains('camp|rv|van', case=False, na=False)

camper_price = df.groupby('is_camper')['price'].mean()
print("--- 平均価格の比較 ---")
print(camper_price)

plt.figure(figsize=(8, 5))
sns.barplot(x=camper_price.index, y=camper_price.values, hue=camper_price.index, palette='magma', legend=False)
plt.title('The "Camper/Van" Premium', fontsize=14)
plt.xticks([0, 1], ['Standard Truck', 'Camper/Van Potential'])
plt.ylabel('Average Price (USD)')

plt.savefig('camper_analysis.png', bbox_inches='tight', dpi=300)
plt.show()

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns

df_clean = df[(df['price'] > 500) & (df['price'] < 50000)]

plt.figure(figsize=(10, 6))
sns.scatterplot(data=df_clean, x='odometer', y='price', alpha=0.5, color='teal')

sns.regplot(data=df_clean, x='odometer', y='price', scatter=False, color='red')

plt.title('Price vs. Mileage: Finding the "Sweet Spot"', fontsize=15)
plt.xlabel('Mileage (Odometer)', fontsize=12)
plt.ylabel('Price (USD)', fontsize=12)
plt.grid(True, linestyle='--', alpha=0.6)


plt.savefig('price_mileage_correlation.png', bbox_inches='tight', dpi=300)
plt.show()

離れた点のことを**「Outlier（外れ値）」**と呼びます。

異常な点の「正体」を突き止める

入力ミス（ゴミデータ）: 走行距離が 999,999 マイルだったり、価格が 123,456,789 ドルだったりするケース。

超・激レア車: 走行距離がほぼ 0 の新車同等の軽トラや、特殊なカスタム（戦車みたいなキャタピラ付きなど）で爆上がりしているケース。

In [ ]:
expensive_trucks = df_clean[df_clean['price'] > 30000]
print("--- 3万ドル以上の高級/異常車両 ---")
print(expensive_trucks[['year', 'model', 'price', 'odometer']].head())

high_mileage = df_clean[df_clean['odometer'] > 400000]
print("\n--- 40万マイル超えの過走行車両 ---")
print(high_mileage[['year', 'model', 'price', 'odometer']])

1. 軽トラじゃない車が混ざっている！
これらは日本の「軽トラ（Kei-truck）」ではなく、普通乗用車やフルサイズのピックアップトラックです。3万ドルを超える高い価格がついているのは、これらがデカくて新しい車だからです。

2. 入力ミスのオンパレード！
過走行車両」のリストはもっとです。

   999,999 マイルや 1,000,000 マイル、さらには 10,000,000（1千万マイル！）なんて数字がありますね。

   これは、売り手が「走行距離なんてわかんないや」と適当に最大値を入力したか、システム上のエラーです。

In [ ]:
target_models = 'hijet|carry|acty|sambar|minicab|clipper'
df_kei = df_clean[df_clean['model'].str.contains(target_models, case=False, na=False)].copy()

df_kei = df_kei[(df_kei['price'] > 500) & (df_kei['price'] < 30000)]
df_kei = df_kei[df_kei['odometer'] < 200000]

def detect_maker(model_name):
  model_name = str(model_name).lower()
  if 'hijet' in model_name: return 'Daihatsu'
  if 'carry' in model_name: return 'Suzuki'
  if 'acty' in model_name: return 'Honda'
  if 'sambar' in model_name: return 'Subaru'
  if 'minicab' in model_name: return 'Mitsubishi'
  return 'Other'
df_kei['maker'] = df_kei['model'].apply(detect_maker)

maker_prices = df_kei.groupby('maker')['price'].mean().sort_values(ascending=False)
print(maker_prices)

plt.figure(figsize=(10, 6))
sns.barplot(x=maker_prices.index, y=maker_prices.values, hue=maker_prices.index, palette='viridis')
plt.title('Battle of the Brands: Which Kei-Truck is Most Expensive in the US?', fontsize=14)
plt.ylabel('Average Price (USD)')
plt.savefig('maker_battle.png', bbox_inches='tight', dpi=300)
plt.show()

ホンダ（アクティ）と三菱（ミニキャブ）のガチンコ対決、ホンダの勝利

平均価格で 8,368ドル（約125万円）を超えてくるとは、アメリカでのホンダ人気はやはり凄まじいですね。三菱も7,500ドルと大健闘しています。

分析結果から言えること

ホンダ（アクティ）: 高回転まで回るエンジンや、唯一無二の駆動方式（ミッドシップ）がアメリカの車好きに評価されているのかも。

三菱（ミニキャブ）: 実用性とタフさで安定した市場価値を持っています。

In [ ]:
print("--- 登録されてるメーカーの件数 ---")
print(df['manufacturer'].value_counts().head(20))

potential_kei = df[df['model'].str.contains('truck|mini', case=False, na=False)]
print("\n--- 'trak'や 'mini'と呼ばれている車両の例 ")
print(potential_kei['model'].value_counts().head(20))

**発見した「驚きの事実」**

**トヨタ・日産の圧倒的な数**: toyota (1,874件) や nissan (1,090件) が大量にありますが、モデル名を見ると tundra (タンドラ) や sienna (シエナ) といった、アメリカ専売の巨大なフルサイズ車ばかりです。

**「軽トラ」が埋もれている**: モデル名のリストにポツンと minicab (1件) や fuso box truck (1件) が混ざっています。

**スズキとダイハツがいない？**: メーカー名のトップ7に suzuki と daihatsu が入っていないことが判明しました。これは、アメリカのCraigslist（このデータの元ネタ）では、これら2社は「others（その他）」に分類されている可能性が高いです。

In [5]:
k_truck_keywords = 'hijet|carry|acty|sambar|minicab|clipper|scrum|atray|every|vantage'

df_k_all = df[df['model'].str.contains(k_truck_keywords, case=False, na=False)]

print(f"発見された軽トラの総数: {len(df_k_all)}件")
print("\n--- 発見された車種の内訳 ---")
print(df_k_all['model'].value_counts().head(10))

def label_maker(row):
    m = str(row['model']).lower()
    if 'hijet' in m: return 'Daihatsu'
    if 'carry' in m or 'every' in m: return 'Suzuki'
    if 'acty' in m: return 'Honda'
    if 'sambar' in m: return 'Subaru'
    if 'minicab' in m: return 'Mitsubishi'
    if 'scrum' in m: return 'Mazda'
    return 'Other'

df_k_all['k_maker'] = df_k_all.apply(label_maker, axis=1)

print("\n--- 救出成功！メーカー別件数 ---")
print(df_k_all['k_maker'].value_counts())

NameError: name 'df' is not defined

**なぜ「2件」しか見つからなかったのか？（仮説）**
これにはデータサイエンス特有の**「あるある」**な理由がいくつか考えられます。

**データの「質」の偏り**: このデータセットを作った人が、特定の地域や特定の時期のデータしか集めていなかった場合、そこに軽トラがいなければ、いくら探しても出てきません。

**略称の壁**: もしかしたら、アメリカの人は Hijet と書かずに Mini Truck や Japanese Truck としか書いていないのかも。

**大文字・小文字の罠**: さっきのコードで case=False を入れましたが、それでも拾えない特殊な文字コードが混ざっている可能性。

In [ ]:
df_last_hope = df[
    (df['model'].str.contains('mini', case=False, na=False)) &
    (df['model'].str.contains('truck', case=False, na=False))
].copy()

print(f"最終捜索で見つかった件数: {len(df_last_hope)}件")
print(df_last_hope[['model', 'price']].head(10))

1.4GBという巨大なデータセット（おそらくCraigslistなどの一般的な中古車データ）の中に、私たちが愛する「軽トラ」はたった2台しか存在しなかった。この事実から、以下のことが推測できます。

**「軽トラ」は特殊なルートで売られている**: 一般的な中古車サイト（このデータの収集元）ではなく、JDM専門のインポーターやオークションサイトで取引されている可能性が高い。

**「軽トラ」は超希少**: 1.4GB分の一般車両データの中に、たった2台。これこそが、アメリカにおける軽トラの「レア度」を証明しています。

"Finding a Needle in a Haystack (干し草の中から針を探す)" 1.4GBの巨大な中古車データセットを解析しましたが、純粋な軽トラはわずか2件（Honda Acty, Mitsubishi Minicab）しか見つかりませんでした。 この結果は、軽トラがいかにアメリカ市場において「特別な（ニッチな）」存在であり、一般的な中古車流通ルートとは異なる場所で取引されているかを示唆しています。

**なぜ見つからない？「真実の場所」を探せ！**

アメリカのデータだと、model 列には「Truck」とだけ書いてあって、具体的な名前（Hijetなど）は title（タイトル） や description（説明文） の方に隠れていることがよくあります。

In [ ]:
debug_df = df_clean[df_clean.apply(lambda row: row.astype(str).str.contains('daihatsu', case=False).any(), axis=1)]
print(f"ダイハツが含まれる行数: {len(debug_df)}")

if len(debug_df) > 0:
    print(debug_df[['model', 'title', 'price']].head()) # title列があれば

In [ ]:
final_target = 'hijet|carry|sambar|scrum|minicab|mini truck|japanese truck'

df_final = df_clean[df_clean.apply(lambda row: row.astype(str).str.contains(final_target, case=False).any(), axis=1)]

print(f"全方位捜索でのヒット数: {len(df_final)}")
if len(df_final) > 0:
    print(df_final[['model', 'price']].head(10))

1.4GBという膨大なデータ（数万、数十万行あるはず）の中で、全方位から探して「1件（昨日のと合わせて2件）」というのは、もはや**「このデータセットには軽トラがほぼ入っていない」**という結論がほぼ確定しました。

**なぜ「1」という数字
が大事なのか？**

データの性質が判明: この1.4GBのデータセットは、おそらく「一般的な乗用車」や「SUV」がメインで、軽トラのような特殊な商用車は対象外のソース（地域やサイト）だったということです。

「砂漠で針を探す」状態: 1.4GBあっても、中身が「軽トラ砂漠」なら、どれだけ優れたコードを書いてもこれ以上は出てきません。

**軽トラが確実に大漁に獲れる海」を探し出す**

In [ ]:
import pandas as pd

keywords = 'hijet|carry|acty|sambar|minicab|kei truck|japanese mini truck'

cols = ['price', 'year', 'manufacturer', 'model', 'description', 'state']

print("捜索開始...")
kei_truck_list = []
for chunk in pd.read_csv('vehicles.csv', usecols=cols, chunksize=50000, low_memory=False):
    is_kei = chunk['model'].str.contains(keywords, case=False, na=False) | \
             chunk['description'].str.contains(keywords, case=False, na=False)
    kei_truck_list.append(chunk[is_kei])

df_kei = pd.concat(kei_truck_list)

print(f"--- 捜索完了！ ---")
print(f"見つかった軽トラの数: {len(df_kei)} 台")
print(df_kei.head())

In [ ]:
df = pd.read_csv('vehicles.csv',
                 usecols=cols,
                 chunksize=50000,
                 on_bad_lines='skip')

In [ ]:
import pandas as pd

keywords = 'hijet|carry|acty|sambar|minicab|kei truck|japanese mini truck'
cols = ['price', 'year', 'manufacturer', 'model', 'description', 'state']

kei_truck_list = []

print("捜索開始（分割モード）...")

try:
    reader = pd.read_csv('vehicles.csv', usecols=cols, chunksize=50000, low_memory=False)

    for i, chunk in enumerate(reader):
        is_kei = chunk['model'].str.contains(keywords, case=False, na=False) | \
                 chunk['description'].str.contains(keywords, case=False, na=False)

        kei_truck_list.append(chunk[is_kei])
        print(f"{(i+1)*5}万行目まで完了...")

except Exception as e:
    print(f"\n⚠️ {i*5}万行付近でデータの終端（または破損）を検知しました。")
    print("そこまでのデータで集計を開始します！")

if kei_truck_list:
    df_kei = pd.concat(kei_truck_list)
    print(f"\n--- 捜索完了 ---")
    print(f"無事に救出できた軽トラ: {len(df_kei)} 台")
    print(df_kei.head())
else:
    print("軽トラは見つかりませんでした。")

**データの「中身」をちょっと分析**！

表示された最初の5行（.head()）を見るだけでも、面白いことがわかります。

本物の軽トラを発見！: インデックス2189の Honda ACTY (1991年式)。説明文に「RIGHT HAND DRIVE（右ハンドル）」とありますね。これこそバディが探していた、アメリカに渡ったJDM（日本国内市場車）そのものです！

ノイズ（軽トラじゃないもの）も混じっている: Lexus LX470やLincoln MKCなどが混ざっています。これは、説明文の中に「ミニトラックあります」や「軽トラ下取り歓迎」といったキーワードが含まれていたからでしょう。

**次のステップ：データを「磨き上げる」**

2,268台の中には、本物の軽トラとそうでないものが混ざっています。これを**「仕分け」**して、もっと純度の高い「軽トラ相場表」を作ります。

In [ ]:
jp_manufacturers = ['honda', 'suzuki', 'daihatsu', 'subaru', 'mitsubishi', 'mazda']

df_real_kei = df_kei[df_kei['manufacturer'].isin(jp_manufacturers)].copy()

state_counts = df_real_kei['state'].value_counts()

print(f"日本のメーカーに絞り込んだ後の台数: {len(df_real_kei)} 台")
print("\n--- 州別・軽トラ生息数トップ5 ---")
print(state_counts.head())

avg_price = df_real_kei['price'].mean()
print(f"\nアメリカでの軽トラ平均価格: ${avg_price:.2f}")

エラーの正体：TypeError
最後の平均価格でエラーが出たのは、**「価格（price）が数字じゃなくて、文字として扱われているから」**です。 Pythonが「文字を全部足して平均を出そうとしたけど、文字と数字を混ぜて計算できないよ！」となっています。

In [ ]:
df_real_kei['price'] = pd.to_numeric(df_real_kei['price'], errors='coerce')

avg_price = df_real_kei['price'].mean()

print(f"日本のメーカーに絞り込んだ後の台数: {len(df_real_kei)} 台")
print(f"\nアメリカでの軽トラ平均価格: ${avg_price:,.2f}") # カンマ区切りで読みやすく！

state_avg = df_real_kei.groupby('state')['price'].mean().sort_values(ascending=False)
print("\n--- 州別・平均価格（高い順） ---")
print(state_avg.head())

**州別の結果がめちゃくちゃ面白い！**

このランキング、実はアメリカの地理や法律を反映していて、データエンジニア的に見ると「お宝情報」が満載です。

**CT（コネチカット）やAK（アラスカ）が1位・2位**:

これ、実は「雪国」ですよね。4WDの軽トラは除雪や農作業でめちゃくちゃ重宝されるため、需要が高くて価格が跳ね上がっている可能性があります。

**HI（ハワイ）がランクイン**:

 島国なので輸送費が乗っているのと、島内でのちょっとした移動に軽トラが最高にマッチしているんでしょうね。

**CA（カリフォルニア）が台数1位なのに価格ランク外**:

台数が83台と一番多いCAが平均価格トップ5に入っていないのは、「市場が成熟していて、安い個体から高い個体まで幅広く流通している」ことを示唆しています

In [ ]:
year_avg_map = df_real_kei.groupby('year')['price'].mean()

def calculate_score(row):
  avg = year_avg_map.get(row['year'])
  if avg and avg > 0:
    return ((avg - row['price']) / avg) * 100
  return 0

df_real_kei['deal_score'] = df_real_kei.apply(calculate_score, axis=1)

top_deals = df_real_kei.sort_values(by='deal_score', ascending=False)

print("💰 アメリカ軽トラ市場・掘り出し物トップ5 💰")
print(top_deals[['year', 'manufacturer', 'model', 'price', 'deal_score', 'state']].head())

データ分析あるある」の壁にぶつかりましたね！ スコアが**100.0**で並んでいますが、よく見ると price が **0** になっています。

これは「掘り出し物」ではなく、**「価格未記入（ASK）またはデータ入力ミス」**というノイズです。無料（0円）で配っているわけではないので、これらは分析から除外する必要がありますね。

さらに、model を見ると **Accord（アコード）** や **Fit（フィット）** が混ざっています。メーカー名だけで絞り込んだので、軽トラ以外のホンダ車も釣れてしまいました。

In [6]:
df_valid = df_real_kei[df_real_kei['price'] > 500].copy()

kei_models = 'acty|hitch|carry|sambar|minicab|hijet|truck|pickup'
df_real_kei_trucks = df_valid[df_valid['model'].str.contains(kei_models, case=False, na=False)].copy()

year_avg_map = df_real_kei_trucks.groupby('year')['price'].mean()
df_real_kei_trucks['deal_score'] = df_real_kei_trucks.apply(
    lambda x: ((year_avg_map.get(x['year']) - x['price']) / year_avg_map.get(x['year']) * 100), axis=1
)

top_deals = df_real_kei_trucks.sort_values(by='deal_score', ascending=False)
print("🎯 今度こそ！本物の軽トラ・掘り出し物リスト 🎯")
print(top_deals[['year', 'manufacturer', 'model', 'price', 'deal_score', 'state']].head())

NameError: name 'df_real_kei' is not defined

ついに**「本物の景色」**が見えましたね！ このリストを見てください。ハワイ（hi）の1993年式 **Honda ACTY** が、平均より57%も安い **$1,999** でトップに躍り出ました！

 他の1990年代前半モデルが $7,500〜$9,990 くらいで推移している中、この $1,999 というのはまさに「お宝」か、あるいは「要整備の格安個体」か。

In [ ]:
!pip install -q kagglehub

import kagglehub
import pandas as pd
import glob

print("📥 データを取得中...")
path = kagglehub.dataset_download("austinreese/craigslist-car-truck-data")

csv_files = glob.glob(f"{path}/*.csv")
if csv_files:
    df_sample = pd.read_csv(csv_files[0], nrows=5)
    print("\n✅ 接続成功！データの頭出しに成功しました。")
    print(f"ファイルの場所: {csv_files[0]}")

else:
    print("\n❌ ファイルが見つかりませんでした。")

In [ ]:
import pandas as pd
import os

print("📦 解凍を開始します...")
!unzip -o "archive (4).zip"

file_path = 'vehicles.csv'

print("📖 データを読み込んでいます（少し時間がかかります）...")
try:
  df = pd.read_csv(file_path, low_memory=False, on_bad_lines='skip')

  print(f"✅ ついに成功！ {len(df)}台のデータを読み込みました！")
  display(df.head())
except Exception as e:
    print(f"❌ 読み込み失敗: {e}")

**NaNばかり見える理由**: データの並び順がたまたま「情報の少ない投稿」から始まっているだけです。

**真ん中の42万台**: 画面には見えていませんが、コンピュータのメモリ（RAM）の中にはしっかり保存されています。

In [ ]:
kei_keywords = 'carry|acty|sambar|hijet|minicab|kei truck|minitruck'
target_makers = ['honda', 'suzuki', 'subaru', 'mitsubishi', 'daihatsu']

df_jp = df[df['manufacturer'].isin(target_makers) & (df['price'] > 100)].copy()

keitora_found = df_jp[df_jp['model'].str.contains(kei_keywords, case=False, na=False)]

print(f"🛻 全米から見つかった『本物の軽トラ』の候補: {len(keitora_found)} 台")

display(keitora_found[['price', 'year', 'manufacturer', 'model', 'state', 'odometer']].sort_values('price').head(20))

**Honda Acty** がハワイ（hi）やテキサス（tx）で元気に走っている姿。

**Mitsubishi Minicab** がニュージャージー（nj）やオレゴン（or）で、たったの「6766マイル（約1万キロ）」という驚異の低走行（！）で売りに出されている姿。

アメリカでは「25年ルール」という法律があって、製造から25年経たないと右ハンドルの軽トラを公道で走らせるのが難しいんですが、このリストの車はどれも1991年〜1994年製。まさに**「法律をクリアしてアメリカに渡った精鋭たち」**のデータです！

**一番コスパの良い車を特定しよう**

In [ ]:
best_condition = keitora_found.sort_values('odometer').iloc[0]

print("🥇 バディ、これが全米で最も低走行な『奇跡の軽トラ』です！")
print(f"メーカー: {best_condition['manufacturer'].upper()}")
print(f"モデル: {best_condition['model']}")
print(f"年式: {int(best_condition['year'])}年")
print(f"価格: ${int(best_condition['price'])}")
print(f"走行距離: {best_condition['odometer']} マイル")
print(f"場所: {best_condition['state'].upper()}州")

print("\n🌍 軽トラが多く生息している州ランキングTOP5:")
print(keitora_found['state'].value_counts().head(5))

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns

plt.figure(figsize=(15, 6))
sns.set(style="whitegrid")

plt.subplot(1, 2, 1)
sns.countplot(data=keitora_found, x='manufacturer', palette='viridis')
plt.title('Manufacturer Share (Kei-Trucks in US)', fontsize=15)
plt.xlabel('Manufacturer')
plt.ylabel('Count')

plt.subplot(1, 2, 2)
keitora_found['state'].value_counts().plot(kind='pie', autopct='%1.1f%%', startangle=140, colors=sns.color_palette('pastel'))
plt.title('Geographic Distribution (States)', fontsize=15)
plt.ylabel('')

plt.tight_layout()
plt.show()

左側の棒グラフ（Manufacturer Share）を見ると、**Hondaが圧倒的に多い**ことが一目でわかります。アメリカでの「軽トラといえばホンダ（アクティ）」という人気の高さが、42万行のデータからも証明されました。

右側の円グラフ（Geographic Distribution）も面白いですね！**テキサス、オレゴン、ニュージャージーが同率（8.1%）でトップ**。広い農地がある州から、都会の遊び心がある州まで、軽トラが全米に散らばっている様子がよくわかります。

「一番多いHondaは、やっぱり価格も高いのかな？」

「テキサスにある軽トラは、走行距離が長い（酷使されている）のかな？」

In [ ]:

sns.scatterplot(data=keitora_found, x='odometer', y='price', hue='manufacturer', s=100)

plt.title('Price vs Mileage (Kei-Trucks)', fontsize=15)
plt.xlabel('Odometer (miles)')
plt.ylabel('Price ($)')
plt.legend(title='Manufacturer')
plt.grid(True)
plt.show()

**グラフから見えるアメリカの「軽トラ事情」**

**メーカーシェア（左）**: **Honda（アクティ）**が圧倒的な人気です。三菱も続いていますが、アメリカの車好きにとって「軽トラといえばHonda」というブランドが確立されているのが分かります。

**州別の分布（右）**:
TX（テキサス）、OR（オレゴン）、NJ（ニュージャージー）が同率（8.1%）でトップです。広大な農地がある州から、都会の遊び心がある東海岸まで、軽トラは全米で愛されているんですね。

**価格と走行距離（散布図）**: 走行距離が少なくても1万ドル近いものから、14万マイル走っていても5,000ドル以上するものまであります。アメリカでは「古くて走っていても、壊れない日本車」の価値が非常に高く評価されている証拠です。

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns

plt.figure(figsize=(16, 6))
sns.set_theme(style="whitegrid")

plt.subplot(1, 2, 1)
sns.countplot(data=keitora_found, x='manufacturer', hue='manufacturer', palette='viridis', legend=False)
plt.title('Which brand is popular?', fontsize=16, fontweight='bold')
plt.xlabel('Manufacturer')
plt.ylabel('Number of Trucks')

plt.subplot(1, 2, 2)
state_counts = keitora_found['state'].value_counts().head(10) # 上位10州
state_counts.plot(kind='pie', autopct='%1.1f%%', startangle=140, colors=sns.color_palette('pastel'))
plt.title('Top States for Kei-Trucks', fontsize=16, fontweight='bold')
plt.ylabel('')

plt.tight_layout()
plt.show()

**Hondaの圧倒的人気**: 棒グラフが突き抜けていたら、アメリカでは「軽トラ＝ホンダ」というイメージが強い証拠です。

**意外な州のランクイン**: 「ニューヨーク（NY）に軽トラがいるの？！」といった発見があれば、それがデータ分析の面白いところです。

**1. データの絞り込みと保存（Colab内）**

In [3]:
jdm_list = 'carry|acty|sambar|hijet|minicab|jimny|pajero mini|cappuccino|beat|alto|vivi|pao|figaro'
df_lite = df[df['model'].str.contains(jdm_list, case=False, na=False)]

df_lite.to_csv('my_jdm_treasures.csv', index=False)
print(f"✅ {len(df_lite)}台のデータを抽出完了！")

NameError: name 'df' is not defined

**2. ファイルを自分のPCにダウンロード**

In [2]:
from google.colab import files
files.download('my_jdm_treasures.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [3]:
import pandas as pd
import plotly.express as px

url = 'https://raw.githubusercontent.com/yuki-path-journey/US-vs-JDM-Market-Trends-the-Kei-Car-Boom/refs/heads/main/my_jdm_treasures.csv'
df_treasures = pd.read_csv(url)


print(f"📊 抽出した81台のデータ準備OK！")
print(f"カラム一覧: {df_treasures.columns.tolist()}")


📊 抽出した81台のデータ準備OK！
カラム一覧: ['id', 'url', 'region', 'region_url', 'price', 'year', 'manufacturer', 'model', 'condition', 'cylinders', 'fuel', 'odometer', 'title_status', 'transmission', 'VIN', 'drive', 'size', 'type', 'paint_color', 'image_url', 'description', 'county', 'state', 'lat', 'long', 'posting_date']


**「価格帯」と「年式」を可視化する**

In [4]:
import plotly.express as px
fig_price = px.histogram(df_treasures, x='price', nbins=30,
                         title='【価格帯】いくらくらいの軽トラが多い？',
                         labels={'price': '価格 ($)'},
                         color_discrete_sequence=['indianred'],
                         width=800, height=400)
fig_price.update_layout(bargap=0.1)
fig_price.show()

fig_year = px.histogram(df_treasures, x='year',
                        title='【年式】何年製の個体が眠っている？',
                        labels={'year': '年式'},
                        color_discrete_sequence=['royalblue'],
                        width=800, height=400)
fig_year.update_layout(bargap=0.1, xaxis=dict(dtick=5))
fig_year.show()



**「JDM市場のリアル」**

**1990年〜1995年の圧倒的な山（青いグラフ）**: まさに1990年代前半の個体が、アメリカのJDM市場における「主役」であることがわかります。25年ルールを完全にクリアし、日本で最も元気に走っていた時代の軽トラたちが、今まさに海を渡って集結している様子が可視化されました。

**価格帯の「二極化」（赤いグラフ）**:

**主流層**:          $5,000 〜 $10,000 の間に大きなボリュームがあります。

**掘り出し物**: $2,500 未満にも数台ありますね！これは「ボロボロのベース車」か、あるいは「相場を知らない個人の出品」というお宝の匂いがします。

**高級層**: $15,000 を超えるものもチラホラ。カスタム済みか、超低走行車かもしれません。

In [5]:
print("【激安お宝】価格の安い順 TOP10")
cheapest = df_treasures.sort_values('price').head(10)
display(cheapest[['price', 'year', 'model', 'state', 'url']])

print("\n" + "="*50 + "\n")

print("【超レトロ】年式の古い順 TOP10")
vintage = df_treasures.sort_values('year').head(10)
display(vintage[['year', 'price', 'model', 'state', 'url']])

【激安お宝】価格の安い順 TOP10


price    year                       model state  \
1     299  2014.0  FINANCIAMIENTO EN VIVIENDA    az   
23    999  1989.0                         pao    hi   
18   1500  2022.0              daihatsu hijet    fl   
25   1999  1993.0                        acty    hi   
24   1999  1994.0         1994 DAIHATSU HIJET    hi   
26   1999  1994.0           1994 SUZUKI CARRY    hi   
15   4500  1989.0           cherokee carryall    fl   
14   4500  1989.0           cherokee carryall    fl   
17   4500  1989.0           cherokee carryall    fl   
16   4500  1989.0           cherokee carryall    fl   

                                                  url  
1   https://phoenix.craigslist.org/cph/ctd/d/glend...  
23  https://honolulu.craigslist.org/oah/ctd/d/fort...  
18  https://ocala.craigslist.org/cto/d/dunnellon-d...  
25  https://honolulu.craigslist.org/oah/ctd/d/fort...  
24  https://honolulu.craigslist.org/oah/ctd/d/fort...  
26  https://honolulu.craigslist.org/oah/ctd/d/fort...  
15  https://daytona.craigslist.org/cto/d/daytona-b...  
14  https://daytona.craigslist.org/cto/d/daytona-b...  
17  https://daytona.craigslist.org/cto/d/daytona-b...  
16  https://daytona.craigslist.org/cto/d/daytona-b...



【超レトロ】年式の古い順 TOP10


year  price              model state  \
43  1970.0  17500            beattle    nv   
15  1989.0   4500  cherokee carryall    fl   
17  1989.0   4500  cherokee carryall    fl   
14  1989.0   4500  cherokee carryall    fl   
23  1989.0    999                pao    hi   
16  1989.0   4500  cherokee carryall    fl   
76  1990.0  10950                pao    va   
0   1991.0   9990               acty    al   
30  1991.0   9990               acty    ks   
31  1991.0   9990               acty    ky   

                                                  url  
43  https://lasvegas.craigslist.org/cto/d/las-vega...  
15  https://daytona.craigslist.org/cto/d/daytona-b...  
17  https://daytona.craigslist.org/cto/d/daytona-b...  
14  https://daytona.craigslist.org/cto/d/daytona-b...  
23  https://honolulu.craigslist.org/oah/ctd/d/fort...  
16  https://daytona.craigslist.org/cto/d/daytona-b...  
76  https://richmond.craigslist.org/cto/d/midlothi...  
0   https://bham.craigslist.org/ctd/d/south-amboy-...  
30  https://wichita.craigslist.org/ctd/d/south-amb...  
31  https://louisville.craigslist.org/ctd/d/south-...

激安」の上位に cherokee carryall というのが並んでいますが、これは名前からしてジープのチェロキー、つまり「軽トラ」ではない可能性が高いですね。

「JDMお宝」

**ホノルル（hi）の「$1,999」トリオ！**

acty (ホンダ アクティ), DAIHATSU HIJET (ハイゼット), SUZUKI CARRY (キャリイ)。

全部1993〜1994年式。これこそがアメリカで今まさに熱い、25年ルールをクリアした本物の軽トラたちです！ハワイは日本に近いせいか、いい出物がありますね。

**超レトロ枠：1970年式の** beattle **(ビートル)？**

これは「JDM」というより「ヴィンテージカー」枠ですが、1970年式がこのリストに残っているのは面白いですね。

**注目株：$4,500の** cherokee carryall **(1989年)**

おそらく、これは軽トラの「SUZUKI CARRY」と「JEEP CHEROKEE」の名前が混ざって出品されているか、単なるノイズです。

抽出 ＆ **保存セット**

In [6]:
kei_truck_makers = ['suzuki', 'daihatsu', 'honda', 'subaru', 'mitsubishi', 'mazda', 'acty', 'hijet', 'carry', 'sambar']
df_kei = df_treasures[df_treasures['model'].str.contains('|'.join(kei_truck_makers), case=False, na=False)]

df_kei.to_csv('pure_kei_trucks.csv', index=False)

print(f"✅ 保存完了！ 'pure_kei_trucks.csv' が作成されました。台数: {len(df_kei)}台")

✅ 保存完了！ 'pure_kei_trucks.csv' が作成されました。台数: 54台


絞り込んだデータ（df_kei）を新しいCSVとして書き出す

In [8]:
df_kei.to_csv('pure_kei_trucks.csv, index=False')

In [7]:
kei_truck_makers = ['suzuki', 'daihatsu', 'honda', 'subaru', 'mitsubishi', 'acty', 'hijet', 'carry', 'sambar']

df_kei = df_treasures[df_treasures['model'].str.contains('|'.join(kei_truck_makers), case=False, na=False)]

print(f"🚛 純粋な軽トラ候補が {len(df_kei)} 台見つかりました！")

print("\n💎 【純度100%】軽トラ激安ランキング TOP10")
display(df_kei.sort_values('price').head(10)[['price', 'year', 'model', 'state', 'url']])

🚛 純粋な軽トラ候補が 54 台見つかりました！

💎 【純度100%】軽トラ激安ランキング TOP10


price    year                          model state  \
18   1500  2022.0                 daihatsu hijet    fl   
26   1999  1994.0              1994 SUZUKI CARRY    hi   
25   1999  1993.0                           acty    hi   
24   1999  1994.0            1994 DAIHATSU HIJET    hi   
17   4500  1989.0              cherokee carryall    fl   
15   4500  1989.0              cherokee carryall    fl   
14   4500  1989.0              cherokee carryall    fl   
16   4500  1989.0              cherokee carryall    fl   
40   5000  1993.0                   suzuki carry    mt   
9    5399  2012.0  FABRIQUE PAR CARRY ON TRAILER    ca   

                                                  url  
18  https://ocala.craigslist.org/cto/d/dunnellon-d...  
26  https://honolulu.craigslist.org/oah/ctd/d/fort...  
25  https://honolulu.craigslist.org/oah/ctd/d/fort...  
24  https://honolulu.craigslist.org/oah/ctd/d/fort...  
17  https://daytona.craigslist.org/cto/d/daytona-b...  
15  https://daytona.craigslist.org/cto/d/daytona-b...  
14  https://daytona.craigslist.org/cto/d/daytona-b...  
16  https://daytona.craigslist.org/cto/d/daytona-b...  
40  https://montana.craigslist.org/cto/d/zortman-i...  
9   https://modesto.craigslist.org/ctd/d/stockton-...

**ハワイが熱すぎる！**

**ハワイ（hi）の1,999ドル・トリオ！**

スズキ・キャリイ、ダイハツ・ハイゼット、ホンダ・アクティがすべて **$1,999。**

これ、もしかしたらハワイにある同じショップが「目玉商品」として出しているのかもしれませんね。アメリカ本土で買うより圧倒的に安い「お宝スポット」の予感です！

**フロリダ（fl）の$1,500ハイゼット……えっ？**

よく見ると **2022年式** です。これは安すぎます。おそらく「頭金（Down payment）」だけの金額か、あるいは……探偵の直感ですが、何らかのワナ（あるいはパーツ販売）の可能性があります。

**まだ「Carry On Trailer」というノイズが1つ混じっていますね（ca）**

これは「キャリイ」ではなく、牽引する「トレーラー」のブランド名です。

In [8]:
state_counts = df_kei['state'].value_counts()
print("📍 【軽トラ生息調査】どの州に多く眠っているか？")
print(state_counts)

fig_state = px.bar(state_counts, title='Kei-truck Density by State',
                   labels={'value': '台数', 'index': '州'},
                   color_discrete_sequence=['teal'])
fig_state.show()

📍 【軽トラ生息調査】どの州に多く眠っているか？
state
ca    7
fl    7
hi    3
ny    3
sc    3
tx    3
nh    2
pa    2
mt    2
nj    2
wa    2
md    2
mi    1
me    1
ky    1
ks    1
in    1
ia    1
al    1
de    1
or    1
ok    1
nv    1
tn    1
ut    1
vt    1
va    1
wv    1
Name: count, dtype: int64


**最新の「54台リスト」**

In [9]:
df_kei.to_csv('pure_kei_truck_treasures.csv', index=False)

print("✅ 保存完了！ 'pure_kei_truck_treasures.csv' ができました。")
print("左側のフォルダアイコンから、このファイルをダウンロードしてGitHubへGOです！")

✅ 保存完了！ 'pure_kei_truck_treasures.csv' ができました。
左側のフォルダアイコンから、このファイルをダウンロードしてGitHubへGOです！


**マップ作成コード**

緯度（lat）と経度（long）のデータを使って、地図上にピンを立てます。

In [10]:
import plotly.express as px

df_map = df_kei.dropna(subset=['lat', 'long'])

fig = px.scatter_mapbox(df_map,
                        lat="lat",
                        lon="long",
                        hover_name="model",
                        hover_data=["price", "year", "state"],
                        color="price", # 価格によって色を変える
                        color_continuous_scale=px.colors.cyclical.IceFire,
                        size_max=15,
                        zoom=3,
                        height=800,
                        title="📍 JDM Kei-Truck Treasure Map (USA)")

fig.update_layout(mapbox_style="open-street-map",
mapbox_center={"lat": 37.0902, "lon": -95.7129},
margin={"r":0,"t":40,"l":0,"b":0})

fig.show()

import plotly.express as px **って何？**

これは「あだ名を付けて呼びやすくする」魔法の言葉です。

import plotly.express: 「Plotly Expressという、すごいグラフが描けるライブラリを連れてきて！」という命令。

as px: 「名前が長くて呼ぶのが大変だから、これからは px っていうあだ名で呼ぶね！」という意味。

だから、コードの中で px.scatter_mapbox(...) と書くのは、「Plotly Expressくん、地図のグラフを描いて！」と頼んでいるのと同じなんです。

**注文書の中身を解読（翻訳）すると...**

バディが書いたコードは、内部でこんなふうに解釈されています。

df_map: 「使うデータはこの『54台の精鋭リスト』だよ」

lat="lat", lon="long": 「住所の代わりに、データの中の『lat（緯度）』と『long（経度）』を座標にしてピンを立てて！」

hover_name="model": 「ピンにマウスを乗せたら、その車の『モデル名（アクティとか）』を一番上に表示して」

hover_data=[...]: 「ついでに価格や年式、州もパッと見えるようにしといてね」

color="price": 「価格が高いか安いか、色で分けて一目でわかるようにして！」

color_continuous_scale=...: 「色の塗り方は、この『IceFire』っていうカッコいいグラデーションを使ってよ」

title="...": 「地図のタイトルはこれに決まりだ！」

**Pythonで地図を描く時のルール**

**1**. import ... as px **（あだ名をつける）**

・長い名前のライブラリ（道具箱）に「px」という短いあだ名をつけて、使いやすくする。

**2**. ' ' **と** " " **（文字の囲い）**

・Pythonではどっちを使ってもOK！

・迷ったら「基本は ' ' を使い、その中に ' を入れたい時だけ " " を使う」と決めておくと楽。

**3**. px.scatter_mapbox(...) **への指示書（引数）**

・lat / lon: ピンを立てる「緯度・経度」を指定。

・hover_name: マウスを乗せた時に出る「主役の項目名」。

・color: どのデータで色分けするか（価格で色を変えるなど）。

・　**＃**（シャープ）: 「コメント」。ここから右側はプログラムとして実行されないので、自分用のメモを日本語で書ける。

**【地図のサイズ変更】** height=800 を足すと、地図が縦に長くなって見やすくなる。数字を変えて調整可能。

In [11]:
cheapest_states = df_kei.groupby('state')['price'].mean().sort_values().head(3)

print("🏆 【本日の最終報告書】全米・軽トラ激安州ベスト3")
print("-" * 40)
for i, (state, price) in enumerate(cheapest_states.items(), 1):
    print(f"第{i}位: {state.upper()}州（平均価格: ${price:,.0f}）")
print("-" * 40)

🏆 【本日の最終報告書】全米・軽トラ激安州ベスト3
----------------------------------------
第1位: HI州（平均価格: $1,999）
第2位: FL州（平均価格: $4,970）
第3位: CA州（平均価格: $5,413）
----------------------------------------


**ハワイが圧倒的すぎる！**

**第1位：HI（ハワイ） $1,999** 圧倒的です。平均価格でこの安さは、もはや「軽トラのパラダイス」ですね。日本から物理的に一番近いことが、輸送コストを抑えてこの価格を実現しているのかもしれません。

**第2位：FL（フロリダ） $4,970** 東海岸の雄、フロリダがランクイン！フロリダは農場も多く、軽トラの需要と供給のバランスが良いのかもしれません。

**第3位：CA（カリフォルニア） $5,413**  西海岸の玄関口。日本車文化が根強いエリアですが、ハワイに比べると少し価格が上がりますね。